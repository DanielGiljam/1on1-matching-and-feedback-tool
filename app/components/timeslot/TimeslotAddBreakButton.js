import React, { Component } from 'react';

export class TimeslotAddBreakButton extends Component {
    render() {
        return (
            <div style={{display:"table"}} onClick={this.props.onClick}>
                <div>
                <svg width="40" height="40">
                    <circle cx="20" cy="20" r="16" stroke="black" strokeWidth="3" fill="transparent" />
                    <line x1="20" x2="20" y1="10" y2="30" stroke="black" strokeWidth="4"/>
                    <line x1="10" x2="30" y1="20" y2="20" stroke="black" strokeWidth="4"/>
                </svg>
                </div>
                <p style={{display:"table-cell", verticalAlign:"middle"}}>Add a break</p>
            </div>
        );
    }
}
