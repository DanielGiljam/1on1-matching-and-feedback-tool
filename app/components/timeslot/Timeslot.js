import React from 'react';
import {TimeslotDrag} from './TimeslotDrag';
import {TimeslotInput} from './TimeslotInput';

const startTime = "10:00";
const endTime = "16:00";
const split = 60;
export class Timeslot extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            available: {
                start: 600,
                end: 960
            }
        };
        this.handleAvailabilityChange = this.handleAvailabilityChange.bind(this);
    }

    handleAvailabilityChange(type, change) {
        var newAvailable = {
            available: {
                start: this.state.available.start,
                end: this.state.available.end
            }
        };
        if (type === 'start') newAvailable.available.start = this.state.available.start + change * split
        if (type === 'end') newAvailable.available.end = this.state.available.end + change * split
        this.setState(newAvailable);
    }

    render() {
        return (
            <div>
                <TimeslotDrag   start={parseTimestamp(startTime)}
                                end={parseTimestamp(endTime)}
                                available={this.state.available}
                                split={split}
                                onAvailabilityChange={this.handleAvailabilityChange} />
                <TimeslotInput start={parseTimestamp(startTime)} end={parseTimestamp(endTime)} split={split} />
            </div>
        );
    }
}

function parseTimestamp(timeString) {
    if (!timeString.match(/^[0-2]?\d:[0-5]\d$/)) return 0;
    var pieces = timeString.split(":");
    return parseInt(pieces[0]) * 60 + parseInt(pieces[1]);
}
