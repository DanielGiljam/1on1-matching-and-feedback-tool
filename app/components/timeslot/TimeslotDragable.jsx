import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TimeslotDragBall from './TimeslotDragBall';
import { parseTimeStamp } from './Timeslot';

class TimeslotDragable extends Component {
  constructor(props) {
    super(props);
    this.handleEndChange = this.handleEndChange.bind(this);
    this.handleStartChange = this.handleStartChange.bind(this);
  }

  handleStartChange(change) {
    this.props.onChange('start', change);
  }
  handleEndChange(change) {
    this.props.onChange('end', change);
  }

  render() {
    const containerStyle = {
      height: this.props.ending - this.props.starting,
      top: this.props.starting,
    };
    let classes = 'dragableContainer';
    if (this.props.dragable !== true) classes += ' unavailable';
    let topDragBall = null;
    let bottomDragBall = null;
    if (this.props.dragable === true) {
      topDragBall = <TimeslotDragBall position="top" onChange={this.handleStartChange} />;
      bottomDragBall = <TimeslotDragBall position="bottom" onChange={this.handleEndChange} />;
    }
    return (
      <div className={classes} style={containerStyle}>
        <p style={{ top: '5px' }}>{parseTimeStamp(Math.round(this.props.start / 5) * 5)}</p>
        <p style={{ bottom: '5px' }}>{parseTimeStamp(Math.round(this.props.end / 5) * 5)}</p>
        {topDragBall}
        {bottomDragBall}
      </div>
    );
  }
}

TimeslotDragable.propTypes = {
  onChange: PropTypes.func, // eslint-disable-line
  ending: PropTypes.number.isRequired,
  starting: PropTypes.number.isRequired,
  dragable: PropTypes.bool.isRequired,
  start: PropTypes.number.isRequired,
  end: PropTypes.number.isRequired,
};

export default TimeslotDragable;
