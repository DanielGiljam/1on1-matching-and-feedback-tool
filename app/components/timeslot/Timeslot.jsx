import React from 'react';
import PropTypes from 'prop-types';
import TimeslotDrag from './TimeslotDrag';
import TimeslotInput from './TimeslotInput';

// timestamp 'HH:MM:SS' to minutes
export function parseMinutes(timeString) {
  if (!timeString.match(/^([0-1]?\d|2[0-3]):[0-5]\d(:|$)/)) return false;
  const pieces = timeString.split(':');
  return (parseInt(pieces[0], 10) * 60) + parseInt(pieces[1], 10);
}
// minutes to timestamp 'HH:MM'
export function parseTimeStamp(minutes) {
  if (minutes < 0 || minutes >= 1440) return false;
  const hours = parseInt(minutes / 60, 10);
  let minutesOver = minutes % 60;
  if (minutesOver < 10) minutesOver = `0${minutesOver}`;
  return `${hours}:${minutesOver}`;
}

/* Component for presenting and editing users availabilities for one day */
class Timeslot extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      // currently set start and end of the availability in minutes
      available: {
        start: this.props.available.start,
        end: this.props.available.end,
      },
      // time of start for the whole day in minutes
      start: this.props.start,
      // time of ending for the whole day in minutes
      end: this.props.end,
      changesMade: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.submitAvailability = this.submitAvailability.bind(this);
    this.handleNavigationClick = this.handleNavigationClick.bind(this);
    this.askIfWantsToSubmitNone = this.askIfWantsToSubmitNone.bind(this);
    this.askForMoreTime = this.askForMoreTime.bind(this);
  }

  // update state when availability changed
  handleChange(to, change) {
    let newStart = this.state.available.start;
    let newEnd = this.state.available.end;
    if (to === 'start') {
      newStart = Math.min(Math.max(newStart + change, this.state.start), this.state.end);
      newStart = Math.min(newStart, newEnd);
      // newEnd = Math.max(newStart, this.state.available.end);
    } else if (to === 'end') {
      newEnd = Math.max(Math.min(newEnd + change, this.state.end), this.state.start);
      newEnd = Math.max(newEnd, newStart);
      // newStart = Math.min(newEnd, this.state.available.start);
    }
    const newObj = { available: { start: newStart, end: newEnd }, changesMade: true };
    this.setState(newObj);
  }

  submitAvailability() {
    let startAvail = Math.round(this.state.available.start / 5) * 5;
    let endAvail = Math.round(this.state.available.end / 5) * 5;
    startAvail = this.askForMoreTime('start', startAvail);
    endAvail = this.askForMoreTime('end', endAvail);
    let submit;
    ({ startAvail, endAvail, submit } = this.askIfWantsToSubmitNone(startAvail, endAvail)); // eslint-disable-line
    if (!submit) return;
    this.props.onSubmit(startAvail, endAvail);
    this.setState({ changesMade: false });
  }

  handleNavigationClick(callback) {
    if (this.state.changesMade) this.submitAvailability();
    callback();
  }

  askIfWantsToSubmitNone(startAvail, endAvail) {
    // cut the availability slot to follow meeting splits
    let startAvailCut = Math.ceil((startAvail - this.state.start) / this.props.split);
    startAvailCut = (startAvailCut * this.props.split) + this.state.start;
    let endAvailCut = Math.floor((endAvail - this.state.start) / this.props.split);
    endAvailCut = (endAvailCut * this.props.split) + this.state.start;
    let submit = true;
    if (endAvail - startAvail > 0 && endAvailCut - startAvailCut <= 0) {
      if (!confirm('You are currently submitting availability with no possible meetings. Do you want to submit?')) { // eslint-disable-line
        submit = false;
      }
    }
    return { endAvail: endAvailCut, startAvail: startAvailCut, submit };
  }

  // ask if the user can come earlier or stay longer
  // if the time to next switch is small enough
  askForMoreTime(type, availability) {
    if (type === 'start') {
      let start = Math.floor((availability - this.state.start) / this.props.split);
      start = (start * this.props.split) + this.state.start;
      const cond1 = availability - start < this.props.split / 2;
      const cond2 = availability - start > 0;
      if (cond1 && cond2) {
        if (confirm(`Could you come ${availability - start} minutes earlier?`)) { // eslint-disable-line
          this.setState({
            available: {
              start,
              end: this.state.available.end,
            },
          });
          return start;
        }
      }
    } else if (type === 'end') {
      let endTime = Math.ceil((availability - this.state.start) / this.props.split);
      endTime = (endTime * this.props.split) + this.state.start;
      const cond1 = endTime - availability < this.props.split / 2;
      const cond2 = endTime - availability > 0;
      if (cond1 && cond2) {
        if (confirm(`Could you stay ${endTime - availability} minutes longer?`)) { // eslint-disable-line
          this.setState({
            available: {
              start: this.state.available.start,
              end: endTime,
            },
          });
          return endTime;
        }
      }
    }
    return availability;
  }

  render() {
    const dateOptions = {
      weekday: 'short',
      day: 'numeric',
      month: 'numeric',
      year: 'numeric',
    };
    const moveToPrevClass = `glyphicon glyphicon-triangle-left${(this.props.onMoveToPrev === undefined) ? ' invisible' : ''}`;
    const moveToNextClass = `glyphicon glyphicon-triangle-right${(this.props.onMoveToNext === undefined) ? ' invisible' : ''}`;
    return (
      <div className="container timeslot-container">
        <p className="date">{this.props.date.toLocaleDateString('en-GB', dateOptions).replace(/\//g, '.')}</p>
        <p className="help-text help-text-center">
          Drag from the arrow to set your availability...
        </p>
        {/* Component for setting by dragging */}
        <TimeslotDrag
          start={this.state.start}
          end={this.state.end}
          available={this.state.available}
          onChange={this.handleChange}
          split={this.props.split}
        />
        <p className="help-text help-text-center">...or type in manually</p>
        {/* Component for setting by typing */}
        <TimeslotInput
          available={this.state.available}
          onChange={this.handleChange}
        />
        {/* Navigation components */}
        <div className="navigation-container">
          <span
            className={moveToPrevClass}
            onClick={() => this.handleNavigationClick(this.props.onMoveToPrev)}
            role="button"
            tabIndex={0}
            onKeyDown={() => {}}
          />
          <button onClick={this.submitAvailability} className="btn btn-lg btn-major">Submit</button>
          <span
            className={moveToNextClass}
            onClick={() => this.handleNavigationClick(this.props.onMoveToNext)}
            role="button"
            tabIndex={0}
            onKeyDown={() => {}}
          />
        </div>
      </div>
    );
  }
}

Timeslot.propTypes = {
  date: PropTypes.objectOf(Date).isRequired,
  start: PropTypes.number.isRequired,
  end: PropTypes.number.isRequired,
  available: PropTypes.shape({
    start: PropTypes.number.isRequired,
    end: PropTypes.number.isRequired,
  }).isRequired,
  split: PropTypes.number.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onMoveToPrev: PropTypes.func,
  onMoveToNext: PropTypes.func,
};

Timeslot.defaultProps = {
  onMoveToNext: undefined,
  onMoveToPrev: undefined,
};

export default Timeslot;
