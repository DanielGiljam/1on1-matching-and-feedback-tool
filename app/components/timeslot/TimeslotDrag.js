import React, { Component } from 'react';
import { TimeslotDragable } from "./TimeslotDragable";

export const splitHeight = 50;

export class TimeslotDrag extends Component {
    render() {
        const containerStyle = {
            position: 'relative',
            height: (this.props.end - this.props.start) / this.props.split * splitHeight
        };
        return (
            <div style={containerStyle}> 
                <TimeslotDragable start={this.props.start} end={this.props.end} split={this.props.split} startingSplit="0" type="unavailable" dragable="false" />
                <TimeslotDragable start={this.props.start + 60} end={this.props.end - 60} split={this.props.split} startingSplit="1" type="available" dragable="true" />
            </div>
        );
    }
}
