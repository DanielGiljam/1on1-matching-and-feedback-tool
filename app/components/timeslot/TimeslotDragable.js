import React, { Component } from 'react';
import {splitHeight} from './TimeslotDrag';
import {TimeslotDragBall} from './TimeslotDragBall';
import {parseTimeStamp} from './Timeslot';

const styles = {
    time: {
        position: 'absolute',
        right: '5px',
        margin: '0px'
    }
};
export class TimeslotDragable extends Component {
    constructor(props) {
        super(props);
        this.handleEndChange = this.handleEndChange.bind(this);
        this.handleStartChange = this.handleStartChange.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }
    
    handleStartChange(change) {
        //check if changed enough
        this.handleChange("start", change);
    }
    handleEndChange(change) {
        this.handleChange("end", change);
    }
    handleChange(type, change) {
        this.props.onChange(type, change / this.props.split);
    }

    render() {
        const containerStyle = {
            background: 'orange',
            position: 'absolute',
            height: (this.props.end - this.props.start) / this.props.split * splitHeight,
            top: this.props.startingSplit * splitHeight,
            width: '100%'
        };
        if (this.props.type === "break") containerStyle.background = "gray";
        else containerStyle.background = "orange";
        var topDragBall = null
        var bottomDragBall = null
        if (this.props.dragable === 'true') {
            topDragBall = <TimeslotDragBall position="top" onChange={this.handleStartChange} onManipulationEnd={this.props.onManipulationEnd} />
            bottomDragBall = <TimeslotDragBall position="bottom" onChange={this.handleEndChange} onManipulationEnd={this.props.onManipulationEnd} />;
        }
        return (
            <div style={containerStyle}>
                <p style={Object.assign({top: '5px'}, styles.time)}>{parseTimeStamp(this.props.start)}</p>
                <p style={Object.assign({bottom: '5px'}, styles.time)}>{parseTimeStamp(this.props.end)}</p>
                {topDragBall}
                {bottomDragBall}
            </div>
        );
    }
}