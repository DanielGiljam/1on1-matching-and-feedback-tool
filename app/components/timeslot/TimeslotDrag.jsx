import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TimeslotDragable from './TimeslotDragable';

export const totalHeight = 300;

// the draggable componet for setting the availability
class TimeslotDrag extends Component {
  constructor(props) {
    super(props);
    this.handleAvailableChange = this.handleAvailableChange.bind(this);
  }

  // change the distance dragged to corresponding minutes and pass the result forward
  handleAvailableChange(to, change) {
    this.props.onChange(to, (change / totalHeight) * (this.props.end - this.props.start));
  }

  render() {
    const containerStyle = {
      position: 'relative',
      height: totalHeight,
      overflow: 'hidden',
    };
    let starting = (this.props.available.start - this.props.start) * totalHeight;
    starting /= (this.props.end - this.props.start);
    let ending = (this.props.available.end - this.props.start) * totalHeight;
    ending /= (this.props.end - this.props.start);
    return (
      <div className="dragContainer" style={containerStyle}>
        {/* undraggable background for the component */}
        <TimeslotDragable
          start={this.props.start}
          end={this.props.end}
          starting={0}
          ending={totalHeight}
          type="break"
          dragable={false}
          split={this.props.split}
          totalHeight={totalHeight}
        />
        {/* the draggable availability box */}
        <TimeslotDragable
          start={this.props.available.start}
          end={this.props.available.end}
          starting={starting}
          ending={ending}
          onChange={this.handleAvailableChange}
          type="available"
          dragable
        />
      </div>
    );
  }
}

TimeslotDrag.propTypes = {
  onChange: PropTypes.func.isRequired,
  end: PropTypes.number.isRequired,
  start: PropTypes.number.isRequired,
  available: PropTypes.shape({
    start: PropTypes.number,
    end: PropTypes.number,
  }).isRequired,
  // length of one meeting
  split: PropTypes.number.isRequired,
};

export default TimeslotDrag;
