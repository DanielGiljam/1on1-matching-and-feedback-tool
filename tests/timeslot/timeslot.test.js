import React from 'react';
import {parseMinutes, parseTimeStamp, Timeslot} from '../../app/components/timeslot/Timeslot';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('parseMinutes works correctly', () => {
    expect(parseMinutes("string")).toBe(false);
    expect(parseMinutes("24:00")).toBe(false);
    expect(parseMinutes("10")).toBe(false);
    expect(parseMinutes("3:63")).toBe(false);
    expect(parseMinutes("9:00")).toBe(540);
    expect(parseMinutes("13:24")).toBe(804);
    expect(parseMinutes("-4:00")).toBe(false);
    expect(parseMinutes("23:59")).toBe(1439);
    expect(parseMinutes("0:00")).toBe(0);
});

test('parseTimeStamp works correctly', () => {
    expect(parseTimeStamp(-1)).toBe(false);
    expect(parseTimeStamp(1440)).toBe(false);
    expect(parseTimeStamp(0)).toBe("0:00");
    expect(parseTimeStamp(512)).toBe("8:32");
    expect(parseTimeStamp(770)).toBe("12:50");
    expect(parseTimeStamp(1439)).toBe("23:59");
    expect(parseTimeStamp(600)).toBe("10:00");
});

test('Timeslot initializes correctly', () => {
    var timeslot = renderer.create(<Timeslot start="8:00" end="12:00" split={40} />);
    let tree = timeslot.toJSON();
    expect(tree).toMatchSnapshot();
    timeslot = renderer.create(<Timeslot start="12:20" end="20:40" split={20} />);
    tree = timeslot.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Timeslot adds breaks correctly', () => {
    const timeslot = shallow(<Timeslot start="8:00" end="12:00" split={30} />);
    timeslot.instance().handleBreakAddition();
    expect(timeslot.state().breaks).toEqual([{start: 510, end: 540}]);
    expect(timeslot).toMatchSnapshot();
    timeslot.instance().handleBreakAddition();
    expect(timeslot.state().breaks).toEqual([{start: 510, end: 540}, {start: 570, end: 600}]);
    expect(timeslot).toMatchSnapshot();
    timeslot.instance().handleBreakAddition();
    timeslot.instance().handleBreakAddition();
    timeslot.instance().handleBreakAddition();
    expect(timeslot.state().breaks).toEqual([{start: 510, end: 540}, {start: 570, end: 600}, {start: 630, end: 660}, {start: 690, end: 720}, {start: 480, end: 510}]);
    expect(timeslot).toMatchSnapshot();
});

describe('Timeslot handles change correctly', () => {
    test("handles availability change", () => {
        const timeslot = shallow(<Timeslot start="10:00" end="12:00" split={20} />);
        timeslot.instance().handleChange("available", 'start', -30);
        expect(timeslot.state().available).toEqual({start: 600, end: 720});
        timeslot.instance().handleChange("available", 'start', 10);
        expect(timeslot.state().available).toEqual({start: 620, end: 720});
        timeslot.instance().handleChange("available", 'start', 9);
        expect(timeslot.state().available).toEqual({start: 620, end: 720});
        timeslot.instance().handleChange("available", "end", 20);
        expect(timeslot.state().available).toEqual({start: 620, end: 720});
        timeslot.instance().handleChange("available", "end", -35);
        expect(timeslot.state().available).toEqual({start: 620, end: 680});
        expect(timeslot).toMatchSnapshot();
        timeslot.instance().handleChange("available", "end", 26);
        expect(timeslot.state().available).toEqual({start: 620, end: 700});
        timeslot.instance().handleChange("available", "start", 100);
        expect(timeslot.state().available).toEqual({start: 700, end: 700});
        expect(timeslot).toMatchSnapshot();
    });
    test("handles break change", () => {
        const timeslot = shallow(<Timeslot start="10:00" end="12:00" split={20} />);
        timeslot.setState({breaks: [{start: 620, end: 640}, {start: 680, end: 700}]});
        timeslot.instance().handleChange("break", "start", -11, 0);
        expect(timeslot.state().breaks).toEqual([{start: 600, end: 640}, {start: 680, end: 700}]);
        timeslot.instance().handleChange("break", "start", -11, 0);
        expect(timeslot.state().breaks).toEqual([{start: 600, end: 640}, {start: 680, end: 700}]);
        timeslot.instance().handleChange("break", "end", -10, 0);
        expect(timeslot.state().breaks).toEqual([{start: 600, end: 640}, {start: 680, end: 700}]);
        timeslot.instance().handleChange("break", "end", 27, 0);
        expect(timeslot.state().breaks).toEqual([{start: 600, end: 660}, {start: 680, end: 700}]);
        expect(timeslot).toMatchSnapshot();
        timeslot.instance().handleChange("break", "end", 40, 1);
        expect(timeslot.state().breaks).toEqual([{start: 600, end: 660}, {start: 680, end: 720}]);
        timeslot.instance().handleChange("break", "end", -60, 1);
        expect(timeslot.state().breaks).toEqual([{start: 600, end: 660}, {start: 680, end: 680}]);
        expect(timeslot).toMatchSnapshot();
    });
});