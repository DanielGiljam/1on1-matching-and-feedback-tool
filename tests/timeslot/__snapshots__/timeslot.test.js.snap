// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeslot handles change correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timeslot
    date={2018-03-02T22:00:00.000Z}
    end="12:00"
    start="10:00"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <link
          href="app/styles/timeslot_style.css"
          rel="stylesheet"
          type="text/css"
        />,
        <p
          className="date"
        >
          Sat, 3.3.18
        </p>,
        <TimeslotDrag
          available={
            Object {
              "end": 600,
              "start": 600,
            }
          }
          end={720}
          onChange={[Function]}
          start={600}
        />,
        <TimeslotInput
          available={
            Object {
              "end": 600,
              "start": 600,
            }
          }
          onChange={[Function]}
        />,
      ],
      "className": "timeslot-picker",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "href": "app/styles/timeslot_style.css",
          "rel": "stylesheet",
          "type": "text/css",
        },
        "ref": null,
        "rendered": null,
        "type": "link",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sat, 3.3.18",
          "className": "date",
        },
        "ref": null,
        "rendered": "Sat, 3.3.18",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "available": Object {
            "end": 600,
            "start": 600,
          },
          "end": 720,
          "onChange": [Function],
          "start": 600,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "available": Object {
            "end": 600,
            "start": 600,
          },
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <link
            href="app/styles/timeslot_style.css"
            rel="stylesheet"
            type="text/css"
          />,
          <p
            className="date"
          >
            Sat, 3.3.18
          </p>,
          <TimeslotDrag
            available={
              Object {
                "end": 600,
                "start": 600,
              }
            }
            end={720}
            onChange={[Function]}
            start={600}
          />,
          <TimeslotInput
            available={
              Object {
                "end": 600,
                "start": 600,
              }
            }
            onChange={[Function]}
          />,
        ],
        "className": "timeslot-picker",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "href": "app/styles/timeslot_style.css",
            "rel": "stylesheet",
            "type": "text/css",
          },
          "ref": null,
          "rendered": null,
          "type": "link",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sat, 3.3.18",
            "className": "date",
          },
          "ref": null,
          "rendered": "Sat, 3.3.18",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "available": Object {
              "end": 600,
              "start": 600,
            },
            "end": 720,
            "onChange": [Function],
            "start": 600,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "available": Object {
              "end": 600,
              "start": 600,
            },
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Timeslot handles change correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timeslot
    date={2018-03-02T22:00:00.000Z}
    end="12:00"
    start="10:00"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <link
          href="app/styles/timeslot_style.css"
          rel="stylesheet"
          type="text/css"
        />,
        <p
          className="date"
        >
          Sat, 3.3.18
        </p>,
        <TimeslotDrag
          available={
            Object {
              "end": 600,
              "start": 600,
            }
          }
          end={720}
          onChange={[Function]}
          start={600}
        />,
        <TimeslotInput
          available={
            Object {
              "end": 600,
              "start": 600,
            }
          }
          onChange={[Function]}
        />,
      ],
      "className": "timeslot-picker",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "href": "app/styles/timeslot_style.css",
          "rel": "stylesheet",
          "type": "text/css",
        },
        "ref": null,
        "rendered": null,
        "type": "link",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sat, 3.3.18",
          "className": "date",
        },
        "ref": null,
        "rendered": "Sat, 3.3.18",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "available": Object {
            "end": 600,
            "start": 600,
          },
          "end": 720,
          "onChange": [Function],
          "start": 600,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "available": Object {
            "end": 600,
            "start": 600,
          },
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <link
            href="app/styles/timeslot_style.css"
            rel="stylesheet"
            type="text/css"
          />,
          <p
            className="date"
          >
            Sat, 3.3.18
          </p>,
          <TimeslotDrag
            available={
              Object {
                "end": 600,
                "start": 600,
              }
            }
            end={720}
            onChange={[Function]}
            start={600}
          />,
          <TimeslotInput
            available={
              Object {
                "end": 600,
                "start": 600,
              }
            }
            onChange={[Function]}
          />,
        ],
        "className": "timeslot-picker",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "href": "app/styles/timeslot_style.css",
            "rel": "stylesheet",
            "type": "text/css",
          },
          "ref": null,
          "rendered": null,
          "type": "link",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sat, 3.3.18",
            "className": "date",
          },
          "ref": null,
          "rendered": "Sat, 3.3.18",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "available": Object {
              "end": 600,
              "start": 600,
            },
            "end": 720,
            "onChange": [Function],
            "start": 600,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "available": Object {
              "end": 600,
              "start": 600,
            },
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Timeslot initializes correctly 1`] = `
<div
  className="timeslot-picker"
>
  <link
    href="app/styles/timeslot_style.css"
    rel="stylesheet"
    type="text/css"
  />
  <p
    className="date"
  >
    Mon, 5.28.18
  </p>
  <div
    className="dragContainer"
    style={
      Object {
        "height": 400,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      className="dragableContainer unavailable"
      style={
        Object {
          "height": 400,
          "top": 0,
        }
      }
    >
      <p
        style={
          Object {
            "top": "5px",
          }
        }
      >
        8:00
      </p>
      <p
        style={
          Object {
            "bottom": "5px",
          }
        }
      >
        12:00
      </p>
    </div>
    <div
      className="dragableContainer"
      style={
        Object {
          "height": 0,
          "top": 0,
        }
      }
    >
      <p
        style={
          Object {
            "top": "5px",
          }
        }
      >
        8:00
      </p>
      <p
        style={
          Object {
            "bottom": "5px",
          }
        }
      >
        8:00
      </p>
      <div
        className="timeslot-dragball"
        draggable="true"
        onDrag={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "height": 12,
            "top": -6,
            "width": 12,
          }
        }
      />
      <div
        className="timeslot-dragball"
        draggable="true"
        onDrag={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "bottom": -6,
            "height": 12,
            "width": 12,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="timeslot-input"
    >
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="8:00"
      />
      <p>
        -
      </p>
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="8:00"
      />
    </div>
  </div>
</div>
`;

exports[`Timeslot initializes correctly 2`] = `
<div
  className="timeslot-picker"
>
  <link
    href="app/styles/timeslot_style.css"
    rel="stylesheet"
    type="text/css"
  />
  <p
    className="date"
  >
    Wed, 2.1.17
  </p>
  <div
    className="dragContainer"
    style={
      Object {
        "height": 400,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      className="dragableContainer unavailable"
      style={
        Object {
          "height": 400,
          "top": 0,
        }
      }
    >
      <p
        style={
          Object {
            "top": "5px",
          }
        }
      >
        12:20
      </p>
      <p
        style={
          Object {
            "bottom": "5px",
          }
        }
      >
        20:40
      </p>
    </div>
    <div
      className="dragableContainer"
      style={
        Object {
          "height": 0,
          "top": 0,
        }
      }
    >
      <p
        style={
          Object {
            "top": "5px",
          }
        }
      >
        12:20
      </p>
      <p
        style={
          Object {
            "bottom": "5px",
          }
        }
      >
        12:20
      </p>
      <div
        className="timeslot-dragball"
        draggable="true"
        onDrag={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "height": 12,
            "top": -6,
            "width": 12,
          }
        }
      />
      <div
        className="timeslot-dragball"
        draggable="true"
        onDrag={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "bottom": -6,
            "height": 12,
            "width": 12,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="timeslot-input"
    >
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="12:20"
      />
      <p>
        -
      </p>
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="12:20"
      />
    </div>
  </div>
</div>
`;
